presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -92,28 +90,28 @@
     private final SymbolAllocator symbolAllocator = new SymbolAllocator();
     private final Metadata metadata;
     private final SqlParser sqlParser;
-    private final CostCalculator costCalculator;
+    private final Lookup lookup;
 
     public LogicalPlanner(Session session,
             List<PlanOptimizer> planOptimizers,
             PlanNodeIdAllocator idAllocator,
             Metadata metadata,
             SqlParser sqlParser,
-            CostCalculator costCalculator)
+            Lookup lookup)
```

Comment:
The `Lookup` I know is created by `Memo` which in turn is created by `IterativeOptimizer` every time its run, and `IterativeOptimizer`-s are members of the `planOptimizers` lists being passed here. What is the relation between  `Lookup` I know and the instance being passed here?

In other words: putting `CostCalculator` behind `Lookup` has much sense to `Rule`-s but I'm not convinced it has it in general. Maybe `CostCalculator` could be usable without `Lookup` (or with a dummy one which handles recursion and caching but not actual lookup).